lib/spack/spack/solver/asp.py

```diff
@@ -823,11 +823,12 @@ def on_model(model):
             print("Statistics:")
             pprint.pprint(self.control.statistics)
 
-        if result.unsolved_specs and setup.concretize_everything:
+        if result.satisfiable and result.unsolved_specs and setup.concretize_everything:
```

Comment:
Thanks, I see this would have been diverting cases where the result was unsatisfiable (which should be handled differently)